controller/internal/enforcer/nfqdatapath/nfq_windows.go

```diff
@@ -0,0 +1,165 @@
+// +build windows
+
+package nfqdatapath
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"syscall"
+	"unsafe"
+
+	"go.aporeto.io/trireme-lib/controller/internal/enforcer/nfqdatapath/afinetrawsocket"
+	"go.aporeto.io/trireme-lib/controller/internal/enforcer/nfqdatapath/nflog"
+	"go.aporeto.io/trireme-lib/controller/internal/windows/frontman"
+	"go.aporeto.io/trireme-lib/controller/pkg/connection"
+	"go.aporeto.io/trireme-lib/controller/pkg/packet"
+	"go.uber.org/zap"
+)
+
+func (d *Datapath) startFrontmanPacketFilter(ctx context.Context, nflogger nflog.NFLogger) error {
+	driverHandle, err := frontman.GetDriverHandle()
+	if err != nil {
+		return err
+	}
+
+	nflogWin := nflogger.(*nflog.NfLogWindows)
+
+	packetCallback := func(packetInfoPtr, dataPtr uintptr) uintptr {
+
+		packetInfo := *(*frontman.PacketInfo)(unsafe.Pointer(packetInfoPtr))
+		packetBytes := (*[1 << 30]byte)(unsafe.Pointer(dataPtr))[:packetInfo.PacketSize:packetInfo.PacketSize]
+
+		var packetType int
+		if packetInfo.Outbound != 0 {
+			packetType = packet.PacketTypeApplication
+		} else {
+			packetType = packet.PacketTypeNetwork
+		}
+
+		// Parse the packet
+		mark := int(packetInfo.Mark)
+		parsedPacket, err := packet.New(uint64(packetType), packetBytes, strconv.Itoa(mark), true)
+
+		parsedPacket.WindowsMetadata = &afinetrawsocket.WindowsPacketMetadata{PacketInfo: packetInfo, IgnoreFlow: false}
+		var processError error
+		var tcpConn *connection.TCPConnection
+		var udpConn *connection.UDPConnection
+		if err != nil {
+			parsedPacket.Print(packet.PacketFailureCreate, d.packetLogs)
+		} else if parsedPacket.IPProto() == packet.IPProtocolTCP {
+			if packetType == packet.PacketTypeNetwork {
+				tcpConn, processError = d.processNetworkTCPPackets(parsedPacket)
+			} else {
+				tcpConn, processError = d.processApplicationTCPPackets(parsedPacket)
+			}
+		} else if parsedPacket.IPProto() == packet.IPProtocolUDP {
+			// process udp packet
+			if packetType == packet.PacketTypeNetwork {
+				udpConn, processError = d.ProcessNetworkUDPPacket(parsedPacket)
+			} else {
+				udpConn, processError = d.ProcessApplicationUDPPacket(parsedPacket)
+			}
+		} else {
+			processError = fmt.Errorf("invalid ip protocol: %d", parsedPacket.IPProto())
+		}
+
+		if processError != nil {
+			if parsedPacket.IPProto() == packet.IPProtocolTCP {
+				d.collectTCPPacket(&debugpacketmessage{
+					Mark:    mark,
+					p:       parsedPacket,
+					tcpConn: tcpConn,
+					udpConn: nil,
+					err:     processError,
+					network: packetType == packet.PacketTypeNetwork,
+				})
+			} else if parsedPacket.IPProto() == packet.IPProtocolUDP {
+				d.collectUDPPacket(&debugpacketmessage{
+					Mark:    mark,
+					p:       parsedPacket,
+					tcpConn: nil,
+					udpConn: udpConn,
+					err:     processError,
+					network: packetType == packet.PacketTypeNetwork,
+				})
+			}
+			// drop packet by not forwarding it
+			return 0
+		}
+
```

Comment:
Please replace with a switch statement for better readability.

Reply:
will simplify code (linux and windows) in another branch and/or after this merge